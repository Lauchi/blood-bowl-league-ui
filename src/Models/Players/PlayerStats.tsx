export class PlayerStats {
    movement: number;
    strength: number;
    agility: number;
    armor: number;

    constructor(movement: number, strength: number, agility: number, armor: number) {
        this.movement = movement;
        this.strength = strength;
        this.agility = agility;
        this.armor = armor;
    }
}