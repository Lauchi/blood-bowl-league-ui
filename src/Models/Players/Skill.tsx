export class Skill {
    skillId: string;

    constructor(skillId: string) {
        this.skillId = skillId;
    }
}