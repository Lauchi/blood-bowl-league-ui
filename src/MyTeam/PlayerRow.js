import React from "react";
import {withTranslation} from "react-i18next";
import TableRow from "@material-ui/core/TableRow";
import TableCell from "@material-ui/core/TableCell";
import SkillCell from "./SkillCell";
import StarPlayerCellContent from "./StarPlayerCellContent";


function PlayerRow(props) {
    const { player, index, t} = props;
    return(
        <TableRow key={index}>
            <TableCell component="th" scope="row">
                {index + 1}
            </TableCell>
            <TableCell>{t("playerTypes." + player.playerTypeId)}</TableCell>
            <TableCell>7</TableCell>
            <TableCell>3</TableCell>
            <TableCell>3</TableCell>
            <TableCell>4</TableCell>
            <TableCell>
                <SkillCell skills={player.skills} startingSkills={player.playerConfig.startingSkills}/>
            </TableCell>
            <TableCell>
                <StarPlayerCellContent starPlayerPoints={player.starPlayerPoints}/>
            </TableCell>
        </TableRow>
    )
}

export default withTranslation()(PlayerRow)