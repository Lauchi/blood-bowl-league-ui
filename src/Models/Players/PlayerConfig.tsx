import {PlayerStats} from "./PlayerStats";
import {Skill} from "./Skill";

export class PlayerConfig {
    playerStats: PlayerStats;
    startingSkills: Skill[];
    playerTypeId: string;

    constructor(playerStats: PlayerStats, startingSkills: Skill[], playerTypeId: string) {
        this.playerStats = playerStats;
        this.startingSkills = startingSkills;
        this.playerTypeId = playerTypeId;
    }
}