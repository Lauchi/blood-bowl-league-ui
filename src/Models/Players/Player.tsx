import {PlayerConfig} from "./PlayerConfig";

export class Player {
    playerPositionNumber: number;
    playerConfig: PlayerConfig;
    skills: string[];
    starPlayerPoints: number;
    playerTypeId: string;
    playerId: string;


    constructor(playerPositionNumber: number, playerConfig: PlayerConfig, skills: string[], starPlayerPoints: number, playerTypeId: string, playerId: string) {
        this.playerPositionNumber = playerPositionNumber;
        this.playerConfig = playerConfig;
        this.skills = skills;
        this.starPlayerPoints = starPlayerPoints;
        this.playerTypeId = playerTypeId;
        this.playerId = playerId;
    }
}