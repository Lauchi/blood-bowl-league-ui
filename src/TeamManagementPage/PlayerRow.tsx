import React, {FunctionComponent} from "react";
import {useTranslation} from "react-i18next";
import TableRow from "@material-ui/core/TableRow";
import TableCell from "@material-ui/core/TableCell";
import SkillCell from "./SkillCell";
import StarPlayerCellContent from "./StarPlayerCellContent";
import {Player} from "../Models/Players/Player";

const PlayerRow:FunctionComponent<{player: Player, index: number}> = ({ player, index }) => {
    const { t } = useTranslation();

    return(
        <TableRow key={index}>
            <TableCell component="th" scope="row">{player.playerPositionNumber}</TableCell>
            <TableCell>{t("playerTypes." + player.playerTypeId)}</TableCell>
            <TableCell>{player.playerConfig.playerStats.movement}</TableCell>
            <TableCell>{player.playerConfig.playerStats.strength}</TableCell>
            <TableCell>{player.playerConfig.playerStats.agility}</TableCell>
            <TableCell>{player.playerConfig.playerStats.armor}</TableCell>
            <TableCell>
                <SkillCell skills={player.skills} startingSkills={player.playerConfig.startingSkills}/>
            </TableCell>
            <TableCell>
                <StarPlayerCellContent starPlayerPoints={player.starPlayerPoints}/>
            </TableCell>
        </TableRow>
    )
}

export default (PlayerRow)