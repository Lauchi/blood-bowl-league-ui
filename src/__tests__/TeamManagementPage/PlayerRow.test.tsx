import React from 'react';
import renderer from 'react-test-renderer';
import PlayerRow from "../../TeamManagementPage/PlayerRow";
import {Player} from "../../Models/Players/Player";
import {PlayerConfig} from "../../Models/Players/PlayerConfig";
import {PlayerStats} from "../../Models/Players/PlayerStats";
import {Skill} from "../../Models/Players/Skill";

describe('PlayerRow', () => {

    const newSkills = ['Block', 'New_skill1', 'New_skill2'];
    const playerConfig = new PlayerConfig(new PlayerStats(7,3,4,8), [new Skill("Block")], 'HU_Blitzer');
    const player = new Player(playerConfig, newSkills, 0, 'HU_Blitzer');

    test('snapshot renders', () => {
        const component = renderer.create(<PlayerRow index={1} player={player}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});