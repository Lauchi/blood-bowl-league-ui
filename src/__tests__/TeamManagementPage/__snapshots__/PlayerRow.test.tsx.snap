// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerRow snapshot renders 1`] = `
<tr
  className="MuiTableRow-root"
>
  <th
    aria-sort={null}
    className="MuiTableCell-root"
    scope="row"
  >
    0
  </th>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    playerTypes.HU_Blitzer
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    7
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    3
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    4
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    8
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    <div
      className="MuiBox-root MuiBox-root-19"
    >
      skills.Block
    </div>
    <div
      className="MuiBox-root MuiBox-root-20"
    >
      skills.New_skill1, skills.New_skill2
    </div>
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root"
  >
    0
     / 
    6
  </td>
</tr>
`;
